util/cpuprofile/cpuprofile.go

```diff
@@ -133,12 +134,15 @@ func (p *parallelCPUProfiler) register(ch ProfileConsumer) {
 		return
 	}
 	p.Lock()
+	isFirst := len(p.cs) == 0
 	p.cs[ch] = struct{}{}
 	p.Unlock()
 
-	select {
-	case p.notifyRegister <- struct{}{}:
-	default:
+	if isFirst {
+		select {
+		case p.notifyRegister <- struct{}{}:
```

Comment:
Generally LGTM. But I'm considering if `notifyRegister` can be eliminated.

Reply:
`notifyRegister` should better to keep it for handling cpu profile request timely for the first consumer.